tools/clang/unittests/HLSLExec/LongVectors.cpp

```diff
@@ -1428,16 +1510,48 @@ class DxilConf_SM69_Vectorized {
     return true;
   }
 
-  template <typename T, OpType OP> void runTest() {
-    WEX::TestExecution::SetVerifyOutput verifySettings(
-        WEX::TestExecution::VerifyOutputSettings::LogOnlyFailures);
-
+  TEST_METHOD_SETUP(methodSetup) {
     // It's possible a previous test case caused a device removal. If it did we
     // need to try and create a new device.
-    if (!D3DDevice || D3DDevice->GetDeviceRemovedReason() != S_OK)
+    if (!D3DDevice || D3DDevice->GetDeviceRemovedReason() != S_OK) {
+      hlsl_test::LogCommentFmt(
+          L"Device was lost: Attempting to create a new D3D12 device.");
       VERIFY_IS_TRUE(
           createDevice(&D3DDevice, ExecTestUtils::D3D_SHADER_MODEL_6_9, false));
+    }
+
+    return true;
+  }
+
+  template <typename T, OpType OP> void runWaveOpTest() {
+    WEX::TestExecution::SetVerifyOutput verifySettings(
+        WEX::TestExecution::VerifyOutputSettings::LogOnlyFailures);
+
+    UINT WaveSize = 0;
+
+    if (OverrideWaveLaneCount > 0) {
+      WaveSize = OverrideWaveLaneCount;
+      hlsl_test::LogCommentFmt(
+          L"Using overridden WaveLaneCount of %d for this test.", WaveSize);
+    } else {
+      D3D12_FEATURE_DATA_D3D12_OPTIONS1 waveOpts;
+      VERIFY_SUCCEEDED(D3DDevice->CheckFeatureSupport(
+          (D3D12_FEATURE)D3D12_FEATURE_D3D12_OPTIONS1, &waveOpts,
```

Comment:
```suggestion
          D3D12_FEATURE_D3D12_OPTIONS1, &waveOpts,
```

Is this cast really necessary?